core/src/components/segment/segment.tsx

```diff
@@ -431,12 +433,74 @@ export class Segment implements ComponentInterface {
     this.checked = current;
   }
 
+  private getSegmentButton = (selector: 'first' | 'last' | 'next' | 'previous' | 'focused'): HTMLIonSegmentButtonElement | null => {
+    const buttons = this.getButtons().filter(button => !button.disabled);
+    const currIndex = buttons.findIndex(button => button === this.checked || button.classList.contains('ion-focused'));
+    switch (selector) {
+      case 'first':
+        return buttons[0];
+      case 'last':
+        return buttons[buttons.length - 1];
+      case 'next':
+        return buttons[currIndex + 1] || buttons[0];
+      case 'previous':
+        return buttons[currIndex - 1 ] || buttons[buttons.length - 1];
+      case 'focused':
+        return this.el.querySelector('ion-segment-button.ion-focused');
+      default:
+        return null;
+    }
+  }
+
+  private onKeyUp = (ev: KeyboardEvent) => {
+    const isRTL = document.dir === 'rtl';
+    let current;
+    switch (ev.key) {
+      case 'ArrowRight':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('previous') : this.getSegmentButton('next');
+        break;
+      case 'ArrowLeft':
+        ev.preventDefault();
+        current = isRTL ? this.getSegmentButton('next') : this.getSegmentButton('previous')
+        break;
+      case 'Home':
+        ev.preventDefault();
+        current = this.getSegmentButton('first');
+        break;
+      case 'End':
+        ev.preventDefault();
+        current = this.getSegmentButton('last');
+        break;
+      case ' ':
+      case 'Enter':
+        ev.preventDefault();
+        current = this.getSegmentButton('focused');
+      default:
+        break;
+    }
+
+    const previous = this.checked || current;
+    if (previous && current) {
+      this.checkButton(previous, current);
```

Comment:
Rather than check the segment button, I think we should focus it and let the user decide whether or not they want to activate it.

This would bring our behavior in line with the MD spec: https://material-components.github.io/material-components-web-catalog/#/component/tabs

Reply:
The following link suggests defaulting to automatic selection unless any associated subsequent actions might cause a network request/delay: https://www.w3.org/TR/wai-aria-practices-1.1/#kbd_selection_follows_focus

I think we are good to go either way though, based on what assumptions we want to make/not make. 

Idea: we could also fairly easily expose a `@Prop() keyboardBehavior: focus | select` on `ion-segment` if we want to leave both possibilities open.

Reply:
Adding the prop was a quick change, so I went ahead and added it with another commit. Let me know if you think it should stay or go. 👌